In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.simplefilter(action="ignore", category=FutureWarning)

## Mengubah parameter default matplotlib
Parameter default matplotlib dapat diubah dengan rcParams sebagai berikut


In [ ]:
from matplotlib import rcParams

rcParams['figure.figsize'] = 12, 4
rcParams['lines.linewidth'] = 3
rcParams['xtick.labelsize'] = 'x-large'
rcParams['ytick.labelsize'] = 'x-large'

## Load dataset

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/E-Commerce_Shipping_Data.csv')
df.head()

ID Warehouse_block  ... Weight_in_gms  Reached.on.Time_Y.N
0   1               D  ...          1233                    1
1   2               F  ...          3088                    1
2   3               A  ...          3374                    1
3   4               B  ...          1177                    1
4   5               C  ...          2484                    1

[5 rows x 12 columns]

## Descriptive Statistics
### Memeriksa kolom-kolom dan nilai hilang dengan `df.info()`

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10999 entries, 0 to 10998
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   ID                   10999 non-null  int64 
 1   Warehouse_block      10999 non-null  object
 2   Mode_of_Shipment     10999 non-null  object
 3   Customer_care_calls  10999 non-null  int64 
 4   Customer_rating      10999 non-null  int64 
 5   Cost_of_the_Product  10999 non-null  int64 
 6   Prior_purchases      10999 non-null  int64 
 7   Product_importance   10999 non-null  object
 8   Gender               10999 non-null  object
 9   Discount_offered     10999 non-null  int64 
 10  Weight_in_gms        10999 non-null  int64 
 11  Reached.on.Time_Y.N  10999 non-null  int64 
dtypes: int64(8), object(4)
memory usage: 1.0+ MB


Kita dapat melihat bahwa:
* Dataframe memiliki total 10999 baris dan 12 kolom
* Dataframe tidak memiliki *null* values di semua kolom
* Target klasifikasi sepertinya adalah kolom `Reached.on.Time_Y.N` dengan tipe data `int64`
* Berarti sisanya adalah *feature*

Dari informasi di atas, kita bisa pisahkan kolom kategorical dan numerical sebagai berikut:

In [ ]:
cats = ['Warehouse_block', 'Mode_of_Shipment', 'Product_importance', 'Gender']
nums = ['ID', 'Customer_care_calls', 'Customer_rating', 'Cost_of_the_Product', 'Prior_purchases', 'Discount_offered', 'Weight_in_gms', 'Reached.on.Time_Y.N' ]

### Sampling untuk memahami data dengan `df.sample()`

Setelah mengetahui kolom apa saja yang ada di dalam dataset, kita lakukan sampling untuk memastikan apakah isi kolomnya sesuai ekspektasi. Biasakan lakukan ini beberapa kali karena seringkali apabila ada keanehan tidak akan keluar pada sampling pertama.

Sampling dapat dilakukan dengan menggunakan `df.sample()`

In [ ]:
df.sample(10)

ID Warehouse_block Mode_of_Shipment  Customer_care_calls  \
7365    7366               B             Ship                    5   
8563    8564               F             Ship                    3   
644      645               A             Ship                    6   
8862    8863               D             Ship                    4   
7697    7698               F             Ship                    3   
3339    3340               B             Ship                    4   
537      538               B             Road                    4   
7498    7499               C             Ship                    5   
10971  10972               B           Flight                    5   
6050    6051               A             Ship                    6   

       Customer_rating  Cost_of_the_Product  Prior_purchases  \
7365                 5                  258                4   
8563                 2                  177                2   
644                  4                  121                5   
8862                 5                  239                5   
7697                 2                  189                3   
3339                 1                  272                2   
537                  4                  228                3   
7498                 5                  249                5   
10971                5                  221                5   
6050                 4                  271                4   

      Product_importance Gender  Discount_offered  Weight_in_gms  \
7365                high      F                 5           4562   
8563                high      M                 9           4648   
644                  low      M                23           1734   
8862                 low      M                 6           1336   
7697                high      M                 1           4997   
3339              medium      F                 5           5252   
537               medium      F                12           3359   
7498              medium      M                 5           1662   
10971                low      M                 8           1718   
6050              medium      M                 8           1867   

       Reached.on.Time_Y.N  
7365                     0  
8563                     0  
644                      1  
8862                     0  
7697                     1  
3339                     0  
537                      1  
7498                     1  
10971                    0  
6050                     0

Beberapa pengamatan:
* Target kita, `Reached.on.Time_Y.N` benar bertipe int64 dengan *range* 0-1, dimana 1 melambangkan kemungkinan 100%

### Statistical summary dengan `df.describe()`
Selanjutnya kita munculkan ringkasan statistik dataframe, baik untuk numerical dan categorical. Hal ini dilakukan untuk mengecek secara singkat akan keberadaan outlier dan karakteristik distribusi untuk `feature` numeric.

In [ ]:
df[nums].describe()

ID  Customer_care_calls  Customer_rating  Cost_of_the_Product  \
count  10999.00000         10999.000000     10999.000000         10999.000000   
mean    5500.00000             4.054459         2.990545           210.196836   
std     3175.28214             1.141490         1.413603            48.063272   
min        1.00000             2.000000         1.000000            96.000000   
25%     2750.50000             3.000000         2.000000           169.000000   
50%     5500.00000             4.000000         3.000000           214.000000   
75%     8249.50000             5.000000         4.000000           251.000000   
max    10999.00000             7.000000         5.000000           310.000000   

       Prior_purchases  Discount_offered  Weight_in_gms  Reached.on.Time_Y.N  
count     10999.000000      10999.000000   10999.000000         10999.000000  
mean          3.567597         13.373216    3634.016729             0.596691  
std           1.522860         16.205527    1635.377251             0.490584  
min           2.000000          1.000000    1001.000000             0.000000  
25%           3.000000          4.000000    1839.500000             0.000000  
50%           3.000000          7.000000    4149.000000             1.000000  
75%           4.000000         10.000000    5050.000000             1.000000  
max          10.000000         65.000000    7846.000000             1.000000

Beberapa pengamatan dari *statistical summary* kolom numerik di atas:
* Kolom Customer_care_calls dan Customer_rating sudah cukup simetris distribusinya (mean dan median tidak berbeda jauh)
* Tidak ada issue pada nilai minimal untuk setiap kolom
* Tidak ditemukan missing values pada setiap kolom
* Tidak ditemukan duplicated pada setiap baris
* Ditemukan nilai yang cukup jauh antara percentile 75% - 100% pada kolom Prior_purchase dan Discount_offered

In [ ]:
df[cats].describe()

Warehouse_block Mode_of_Shipment Product_importance Gender
count            10999            10999              10999  10999
unique               5                3                  3      2
top                  F             Ship                low      F
freq              3666             7462               5297   5545

Beberapa pengamatan dari *statistical summary* kolom kategorikal di atas:
* Mayoritas gender adalah F dengan frekuensi 5545
* Jenis pengiriman tertinggi dengan 'Ship', yaitu sebanyak 7462 atau 68%

### Value Counting
Kita dapat memastikan pengamatan mengenai kolom-kolom *categorical* di atas dengan value counting

In [ ]:
for col in cats:
    print(f'''Value count kolom {col}:''')
    print(df[col].value_counts())
    print()

Value count kolom Warehouse_block:
F    3666
D    1834
C    1833
B    1833
A    1833
Name: Warehouse_block, dtype: int64

Value count kolom Mode_of_Shipment:
Ship      7462
Flight    1777
Road      1760
Name: Mode_of_Shipment, dtype: int64

Value count kolom Product_importance:
low       5297
medium    4754
high       948
Name: Product_importance, dtype: int64

Value count kolom Gender:
F    5545
M    5454
Name: Gender, dtype: int64

